index.js

```diff
@@ -40,27 +53,25 @@ function retrieve(program, callback){
 
 
   function wrappedCb(err){
-    if(err) errs.push(err);
+    if(err) output.errors.push(err);
 
     try{
       fs.removeSync(scratchSpace);
     }catch(e){
-      if(e) errs.push(e);
+      if(e) output.errors.push(e);
     }
 
-    if(errs.length){
-      logger.error('Encountered %d error%s whilst retrieving.', errs.length, errs.length > 1 ? 's' : '');
-      errs.forEach(function(v){
-        logger.error(v);
-      });
-      if(!callback) throw errs.join('\n');
-    }
+    output.endTime = Date.now();
+
+    if(output.errors.length && !callback) throw output.errors.join('\n');
 
-    if(callback) callback(errs, processedRecords);
+    if(callback) callback(output);
   }
 
+
   if(!program.file) return wrappedCb(new Error('Must provide a metadata file with the -f option.'));
 
+  var monitoringMode = !program.bucket && !program.directory;
```

Comment:
What is `monitoringMode`, exactly?  It checks to see if there are changes, but doesn't actually push files up to S3 or local directory?  Seems worth noting there is such a thing in [`program`](https://github.com/cfpb/grasshopper-retriever/pull/29/files#diff-0bca579da47322e18a757530d945e94dR18) and/or [Usage](https://github.com/cfpb/grasshopper-loader#usage) section of the `README.md`.